scripts/generate_metric_enums.py

```diff
@@ -0,0 +1,144 @@
+            if match:
+                optimizer_type = match[1] if match[1] else match[2]
+                if optimizer_type == "INVALID":
+                    continue
+                optimizer_types.append(optimizer_type)
+
+header = """//-------------------------------------------------------------------------
+//                         DuckDB
+//
+//
+// duckdb/common/enums/metrics_type.hpp
+// 
+// This file is automatically generated by scripts/generate_metric_enums.py
+// Do not edit this file manually, your changes will be overwritten
+//-------------------------------------------------------------------------\n
+"""
+
+typedefs = """struct MetricsTypeHashFunction {
+	uint64_t operator()(const MetricsType &index) const {
+		return std::hash<uint8_t>()(static_cast<uint8_t>(index));
+	}
+};
+
+typedef unordered_set<MetricsType, MetricsTypeHashFunction> profiler_settings_t;
+typedef unordered_map<MetricsType, Value, MetricsTypeHashFunction> profiler_metrics_t;
+
+"""
+
+# Write the metric type header file
+with open(metrics_header_file, "w") as f:
+    f.write(header)
+
+    f.write('#pragma once\n\n')
+    f.write('#include "duckdb/common/constants.hpp"\n')
+    f.write('#include "duckdb/common/unordered_set.hpp"\n\n')
+
+    f.write("namespace duckdb {\n\n")
+
+    f.write("enum class MetricsType : uint8_t {\n")
+
+    for metric in metrics:
+        f.write(f"    {metric},\n")
+
+    for metric in optimizer_types:
+        f.write(f"    OPTIMIZER_{metric}_TIMING,\n")
+
+    f.write("};\n\n")
+
+    f.write(typedefs)
+
+    f.write('profiler_settings_t GetAllOptimizerMetrics();\n')
+    f.write('MetricsType GetOptimizerMetricFromOptimizerType(OptimizerType type);\n\n')
+
+    f.write("} // namespace duckdb\n")
+
+# Write the metric_type.cpp file
+with open(metrics_cpp_file, "w") as f:
+    f.write(header)
+
+    f.write('#include "duckdb/common/enums/metric_type.hpp"\n')
+    f.write('#include "duckdb/common/enums/optimizer_type.hpp"\n\n')
+    f.write("namespace duckdb {\n\n")
+
+    f.write('profiler_settings_t GetAllOptimizerMetrics() {\n')
+    f.write(f"    return {{\n")
+
+    for metric in optimizer_types:
+        f.write(f"        MetricsType::OPTIMIZER_{metric}_TIMING,\n")
+    f.write("    };\n")
+    f.write("}\n\n")
+
+    f.write('MetricsType GetOptimizerMetricFromOptimizerType(OptimizerType type) {\n')
+    f.write('    switch(type) {\n')
+
+    for metric in optimizer_types:
+        f.write(f"        case OptimizerType::{metric}:\n")
+        f.write(f"            return MetricsType::OPTIMIZER_{metric}_TIMING;\n")
+
+    f.write('       default:\n')
+    f.write('            throw InternalException("OptimizerType %s cannot be converted to a MetricType", '
+            'EnumUtil::ToString(type));\n')
+    f.write('    };\n')
+    f.write('}\n\n')
+
+    f.write("} // namespace duckdb\n")
+
+# Run the generate_enum_util.py script to update the enums.hpp file
+os.system("python generate_enum_util.py")
```

Comment:
I'm unsure if this will work. I need to run `python3` specifically. Clang tidy will complain about this in the CI, so I suggest people run this manually.